irobotclient/test/test_main.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+from unittest.mock import MagicMock
+
+import requests
+import os
+
+import irobotclient.main
+
+
+class TestMain(unittest.TestCase):
+
+    test_data_dir = os.path.join(os.path.dirname(__file__), '..', '..', 'testdata/')
+    output_dir = os.path.join(os.path.dirname(__file__), '..', '..', 'testdir/')
+
+    def setUp(self):
+        self._old_request_head = requests.head
+        self._response = requests.Response()
+        requests.head = MagicMock(spec=requests.head)
```

Comment:
It would be good not to be mocking low level request buisness at this level, which is not concerned with the raw request objects. Use of dummy `Request` instances and DI would enable this.